leader-election-impl/src/main/java/com/palantir/paxos/VerifyingPaxosStateLog.java

```diff
@@ -53,6 +56,7 @@
     private final PaxosStateLog<V> currentLog;
     private final PaxosStateLog<V> experimentalLog;
     private final Persistable.Hydrator<V> hydrator;
+    private final CompletableFuture<Long> firstEntry = new CompletableFuture<>();
```

Comment:
discussed offline: I'd probably write `AtomicReference<OptionalLong>`, though this is fine as well